lib/nostrum/struct/channel.ex

```diff
@@ -240,47 +575,107 @@ defmodule Nostrum.Struct.Channel do
           name: name
         }
   @typedoc """
-  A `Nostrum.Struct.Channel` that represents a channel in a guild
+  Guild channel types
   """
+  @typedoc deprecated: "See t.0"
   @type guild_channel ::
           guild_text_channel
           | guild_voice_channel
-          | channel_category
+          | guild_category_channel
 
   @typedoc """
-  A `Nostrum.Struct.Channel` that represents a text channel
+  All valid text channels.
   """
+  @typedoc deprecated: "See t.0"
   @type text_channel ::
           guild_text_channel
           | dm_channel
           | group_dm_channel
 
   @typedoc """
   A `Nostrum.Struct.Channel` that represents a voice channel
+
   """
+  @typedoc deprecated: "See t.0"
   @type voice_channel :: guild_voice_channel
 
+  @typedoc """
+  All valid channel types.
+  """
   @type t ::
           guild_text_channel
           | dm_channel
           | guild_voice_channel
           | group_dm_channel
-          | channel_category
+          | guild_category_channel
+          | guild_news_channel
+          | guild_store_channel
+          | guild_news_thread_channel
+          | guild_public_thread_channel
+          | guild_private_thread_channel
+          | guild_stage_voice_channel
 
-  @doc ~S"""
-  Formats a `Nostrum.Struct.Channel` into a mention
+  @doc """
+  Convert a channel into a mention.
+
+
+  Handles the conversion of a `Nostrum.Struct.Channel` into the required format to _mention_ the channel within a message.
+
+  ## Parameters
+
+    - channel: `t:Nostrum.Struct.Channel.t/0`
 
   ## Examples
 
-  ```Elixir
-  iex> channel = %Nostrum.Struct.Channel{id: 381889573426429952}
-  ...> Nostrum.Struct.Channel.mention(channel)
-  "<#381889573426429952>"
+  ```elixir
+      > Nostrum.Cache.ChannelCache.get(381889573426429952)
+      > |> Nostrum.Struct.Channel.mention()
+      "<#381889573426429952>"
+
+      > Nostrum.GuildCache.get(81384788765712384)
+      > |> Nostrum.Struct.Channel.mention()
+      "<#Invalid Channel>"
   ```
+
   """
   @spec mention(t) :: String.t()
+  def mention(channel)
   def mention(%__MODULE__{id: id}), do: "<##{id}>"
 
+  def mention(_any), do: "<#Invalid Channel>"
```

Comment:
Hm. I get the idea of this, but an explicit error here would be better (with a MatchError from the code that we had beforehand). If no channel is passed in, it explodes, and that's a good thing, in my opinion.

Reply:
No, I didn't mean a banged version, I meant that we keep at a single function that raises at call time if you pass in something bad.

Mentioning `<#Invalid Channel>` by accident would be a major user mistake in the code, as in: the user passing in a wrong type to our function. Dialyzer would already warn about that, and I think it's perfectly reasonable and good to crash with a function clause error here, as otherwise this error will just strobble around in there silently.